File: hcx-pipeline-jobs/claims-job/src/main/scala/org/swasth/dp/claims/functions/ClaimsProcessFunction.scala
package org.swasth.dp.claims.functions

import org.apache.flink.api.common.typeinfo.TypeInformation
import org.slf4j.LoggerFactory
import org.swasth.dp.claims.task.ClaimsConfig
import org.swasth.dp.core.function.{BaseDispatcherFunction, ValidationResult}

import java.util

class ClaimsProcessFunction(config: ClaimsConfig)(implicit val stringTypeInfo: TypeInformation[String])
  extends BaseDispatcherFunction(config) {

  private[this] val logger = LoggerFactory.getLogger(classOf[ClaimsProcessFunction])

  override def validate(event: util.Map[String, AnyRef]): ValidationResult = {
    // TODO: Add domain specific validations
    ValidationResult(true, None)
  }

}

File: hcx-pipeline-jobs/claims-job/src/main/java/org/swasth/dp/claims/functions/ClaimsProcessFunction.java
package org.swasth.dp.claims.functions;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.swasth.dp.claims.task.ClaimsConfig;
import org.swasth.dp.core.function.BaseDispatcherFunction;
import org.swasth.dp.core.function.ValidationResult;

import java.util.Map;

public class ClaimsProcessFunction extends BaseDispatcherFunction {

    private Logger logger = LoggerFactory.getLogger(ClaimsProcessFunction.class);
    private ClaimsConfig config;

    public ClaimsProcessFunction(ClaimsConfig config) {
        super(config);
        this.config = config;
    }

    @Override
    public ValidationResult validate(Map<String, Object> event) {
        // TODO: Add domain specific validations
        return new ValidationResult(true, null);
    }

}
